File: server/src/models/PlaylistItem.ts
import { Model, DataTypes } from 'sequelize';
import sequelize from '../db/connection';
import Room from './Room';

export enum PlayStatus {
  PENDING = 'pending',
  PLAYING = 'playing',
  PAUSED = 'paused',
  FINISHED = 'finished'
}

class PlaylistItem extends Model {
  declare id: number;
  declare roomId: number;
  declare title: string;
  declare orderIndex: number;
  declare playStatus: PlayStatus;
  declare createdTime: Date;
}

PlaylistItem.init({
  id: {
    type: DataTypes.INTEGER,
    primaryKey: true,
    autoIncrement: true
  },
  roomId: {
    type: DataTypes.INTEGER,
    allowNull: false,
    references: {
      model: Room,
      key: 'id'
    }
  },
  title: {
    type: DataTypes.STRING(255),
    allowNull: false
  },
  orderIndex: {
    type: DataTypes.INTEGER,
    allowNull: false
  },
  playStatus: {
    type: DataTypes.ENUM(...Object.values(PlayStatus)),
    allowNull: false,
    defaultValue: PlayStatus.PENDING
  },
  createdTime: {
    type: DataTypes.DATE(3),
    allowNull: false,
    defaultValue: DataTypes.NOW
  }
}, {
  sequelize,
  modelName: 'PlaylistItem',
  tableName: 'playlist_items',
  timestamps: false
});

export default PlaylistItem; 

File: server/internal/models/playlist_item.go
package models

import (
	"time"

	"gorm.io/gorm"
)

// PlayStatus represents the status of a playlist item
type PlayStatus string

const (
	PlayStatusNew      PlayStatus = "new"
	PlayStatusPlaying  PlayStatus = "playing"
	PlayStatusFinished PlayStatus = "finished"
)

// PlaylistItem represents an item in the playlist
type PlaylistItem struct {
	ID          uint           `gorm:"primaryKey;autoIncrement" json:"id"`
	RoomID      uint           `gorm:"not null;index" json:"roomId"`
	Title       string         `gorm:"type:varchar(255);not null" json:"title"`
	OrderIndex  int            `gorm:"not null" json:"orderIndex"`
	PlayStatus  PlayStatus     `gorm:"type:varchar(20);not null;default:'new'" json:"playStatus"`
	CreatedTime time.Time      `gorm:"not null;autoCreateTime:milli" json:"createdTime"`
	DeletedAt   gorm.DeletedAt `gorm:"index" json:"-"`

	// Associations
	Room         *Room          `gorm:"foreignKey:RoomID" json:"room,omitempty"`
	VideoSources []VideoSource  `gorm:"foreignKey:PlaylistItemID" json:"videoSources,omitempty"`
}

// TableName specifies the table name for PlaylistItem model
func (PlaylistItem) TableName() string {
	return "playlist_items"
}
